libs/datamodel/core/tests/attributes/field_name_clash.rs

```diff
@@ -0,0 +1,113 @@
+use crate::{with_header, Provider};
+use expect_test::expect;
+use indoc::indoc;
+
+#[test]
+fn naming_a_field_to_a_generated_id_name_should_error() {
+    let dml = with_header(
+        indoc! {r#"
+        model User {
+          a           Int
+          b           Int
+          a_b         Int
+
+          @@id([a, b])
+        }
+    "#},
+        Provider::Postgres,
+        &[],
+    );
+
+    let error = datamodel::parse_schema(&dml).map(drop).unwrap_err();
+
+    let expectation = expect![[r#"
+        [1;91merror[0m: [1mError validating model "User": The field `a_b` clashes with the `@@id` attribute. Please remove the name clash by providing a custom index name: @@id([..], name: "custom_name")[0m
```

Comment:
The error message references index instead of primary key name. If its not too much work this should be changed.